core/common/src/geometry/Placement.ts

```diff
@@ -1,163 +1,163 @@
+ * @public
+ */
+export type AxisAlignedBox3d = Range3d;
+
+/** The properties of a Range3d.
+ * @public
+ */
+export type AxisAlignedBox3dProps = Range3dProps;
+
+/** A bounding box aligned to the orientation of a 3d Element
+ * @public
+ */
+export type ElementAlignedBox3d = Range3d;
+
+/** A bounding box aligned to the orientation of a 2d Element
+ * @public
+ */
+export type ElementAlignedBox2d = Range2d;
+
+/** A bounding box aligned to a local coordinate system
+ * @public
+ */
+export type LocalAlignedBox3d = Range3d;
+
+/** The placement of a GeometricElement3d. This includes the origin, orientation, and size (bounding box) of the element.
+ * All geometry of a GeometricElement are relative to its placement.
+ * @public
+ */
+export class Placement3d implements Placement3dProps {
+  public constructor(public origin: Point3d, public angles: YawPitchRollAngles, public bbox: ElementAlignedBox3d) { }
+  /** Get the rotation from local coordinates of this placement to world coordinates. */
+  public get rotation(): Matrix3d { return this.angles.toMatrix3d(); }
+  /** Get the transform from local coordinates of this placement to world coordinates. */
+  public get transform(): Transform { return Transform.createOriginAndMatrix(this.origin, this.rotation); }
+
+  /** Create a new Placement3d from a Placement3dProps. */
+  public static fromJSON(json?: Placement3dProps): Placement3d {
+    const props: any = json ? json : {};
+    return new Placement3d(Point3d.fromJSON(props.origin), YawPitchRollAngles.fromJSON(props.angles), Range3d.fromJSON<ElementAlignedBox3d>(props.bbox));
+  }
+
+  /** Get the 8 corners, in world coordinates, of this placement. */
+  public getWorldCorners(out?: Frustum): Frustum {
+    const frust = Frustum.fromRange(this.bbox, out);
+    frust.multiply(this.transform);
+    return frust;
+  }
+
+  /** Set the contents of this Placement3d from another Placement3d */
+  public setFrom(other: Placement3d) {
+    this.origin.setFrom(other.origin);
+    this.angles.setFrom(other.angles);
+    this.bbox.setFrom(other.bbox);
+  }
+
+  /** Determine whether this Placement3d is valid. */
+  public get isValid(): boolean { return !this.bbox.isNull && Math.max(this.origin.maxAbs(), this.bbox.maxAbs()) < Constant.circumferenceOfEarth; }
+
+  /** Calculate the axis-aligned bounding box for this placement. */
+  public calculateRange(): AxisAlignedBox3d {
+    const range = new Range3d();
+    if (!this.isValid)
+      return range;
+
+    this.transform.multiplyRange(this.bbox, range);
+
+    // low and high are not allowed to be equal
+    range.ensureMinLengths();
+    return range;
+  }
+
+  /** Multiply the Transform of this Placement3d by the specified *other* Transform.
+   * @throws [[IModelError]] if the Transform is invalid for a GeometricElement3d.
+   */
+  public multiplyTransform(other: Transform): void {
+    const transform = other.multiplyTransformTransform(this.transform);

```

Comment:
just curious. Why change this from post-multiple to pre-multiply?

Reply:
It was a bug, as discussed with Earlin and Brien - it never worked correctly if `other` had translation and `this.transform` had rotation (e.g., in your move tool). We want to apply a transform to the element's placement, not apply the element's placement to a transform.